src/Classes/TradeQuery.lua

```diff
@@ -0,0 +1,897 @@
+-- Path of Building
+--
+-- Module: Trade Query
+-- Provides PoB Trader pane for interacting with PoE Trade
+--
+
+local dkjson = require "dkjson"
+
+local get_time = os.time
+local t_insert = table.insert
+local t_remove = table.remove
+local m_max = math.max
+local m_min = math.min
+local m_ceil = math.ceil
+
+local baseSlots = { "Weapon 1", "Weapon 2", "Helmet", "Body Armour", "Gloves", "Boots", "Amulet", "Ring 1", "Ring 2", "Belt", "Flask 1", "Flask 2", "Flask 3", "Flask 4", "Flask 5" }
+
+-- map of PoE.Ninja Currency converstion endpoint to GGG's Trade name spec
+local currencyConversionTradeMap = { }
+currencyConversionTradeMap["Orb of Alteration"] = "alt"
+currencyConversionTradeMap["Orb of Fusing"] = "fusing"
+currencyConversionTradeMap["Orb of Alchemy"] = "alch"
+currencyConversionTradeMap["Chaos Orb"] = "chaos"
+currencyConversionTradeMap["Gemcutter's Prism"] = "gcp"
+currencyConversionTradeMap["Exalted Orb"] = "exalted"
+currencyConversionTradeMap["Chromatic Orb"] = "chrome"
+currencyConversionTradeMap["Jeweller's Orb"] = "jewellers"
+currencyConversionTradeMap["Engineer's Orb"] = "engineers"
+currencyConversionTradeMap["Infused Engineer's Orb"] = "infused-engineers-orb"
+currencyConversionTradeMap["Orb of Chance"] = "chance"
+currencyConversionTradeMap["Cartographer's Chisel"] = "chisel"
```

Comment:
It would be more involved than you think. I started that way, then arrived at this as it was a much smaller table that's needed and a lot easier to actually maintain.